solr/solrj/src/java/org/apache/solr/client/solrj/cloud/autoscaling/Policy.java

```diff
@@ -288,10 +287,11 @@ public boolean equals(Object o) {
     return getClusterPreferences().equals(policy.getClusterPreferences());
   }
 
-//  @Override
-//  public int hashCode() {
-//    throw new UnsupportedOperationException("TODO unimplemented");
-//  }
+  //TODO: Uncertain about this one
+  @Override
+  public int hashCode() {
+    return Objects.hash(zkVersion);
```

Comment:
`equals()` compares `policies`, `clusterPolicy` and `clusterPreferences`, but does not compare `zkVersion`.
`hashCode()` should not use `zkVersion` in hash computation.

Reply:
Done. In the interests of speed, would just returning just the hash of getPolicies(). Even though in this case that's just the member var, future-proofing if that changes seems wise.

Reply:
Agreed. Given that `equals()` calls the getter, wise to do the same in `hashCode()`.